### ISSUE TITLE
Helm chart bundled with the release do not pin operator version and use testing image registry

### ISSUE THREAD

**Issue by invidian** [istio/istio#26920]
**Bug description**

It seems that the helm chart bundled with the release does not pin operator image version and use `latest`, which may break the reproducibility of installations.

Also, `hub` used in the release is `gcr.io/istio-testing`, which suggests, that it might actually pull some tests images (with combination with `latest` tag), which seems to be confusing (https://github.com/kinvolk/lokomotive/pull/686#issuecomment-682671543). 

[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

Images in the release bundle should be pinned to specific versions to ensure reproducibility.

**Steps to reproduce the bug**
```sh
https://github.com/istio/istio/releases/download/1.7.0/istio-1.7.0-linux-amd64.tar.gz
tar zxvf istio-1.7.0-linux-amd64.tar.gz
grep -E '^(hub|tag)' istio-1.7.0/manifests/charts/istio-operator/values.yaml
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

`1.7.0`

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

**Comment by howardjohn** [comment]
cc @ostromart @sdake

**Comment by surajssd** [comment]
This is still an issue:

```bash
wget https://github.com/istio/istio/releases/download/1.7.3/istio-1.7.3-linux-amd64.tar.gz
tar zxvf istio-1.7.3-linux-amd64.tar.gz 
grep -E '^(hub|tag)' istio-1.7.3/manifests/charts/istio-operator/values.yaml
```

**Comment by sdake** [comment]
@howardjohn thanks for the triage.

@invidian our apologies for the lag time in resolving what appears to be a simple problem. It is actually more complex than it appears.

@brian-avery this requires changes to release builder. Willing to take this on?

**Comment by invidian** [comment]
Not stale.

**Comment by surajssd** [comment]
A more parameterised way to test if this is still a problem:

```bash
export istio_version=1.8.2
curl -LOs https://github.com/istio/istio/releases/download/$istio_version/istio-$istio_version-linux-amd64.tar.gz
tar zxf istio-$istio_version-linux-amd64.tar.gz 
grep -E '^(hub|tag)' istio-$istio_version/manifests/charts/istio-operator/values.yaml
```

**Comment by howardjohn** [comment]
cc @jacob-delgado I think this is related to the changes you made?

**Comment by jacob-delgado** [comment]
> cc @jacob-delgado I think this is related to the changes you made?

I don't believe so. The instructions for 1.7.0 found at https://istio.io/v1.7/docs/setup/install/operator/

say to do

```shell
$ helm template manifests/charts/istio-operator/ \
  --set hub=docker.io/istio \
  --set tag=1.7.4 \
  --set operatorNamespace=istio-operator \
  --set watchedNamespaces=istio-system | kubectl apply -f -
```

But I agree this is a poor user experience. I'll try and get this fixed for 1.9.0.

**Comment by howardjohn** [comment]
Yeah I meant your fix may apply here as well, not that your change broke
it. I agree it was always not ideal. Thanks!
